src/main/java/org/cactoos/scalar/Xor.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017-2020 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos.scalar;
+
+import org.cactoos.Scalar;
+import org.cactoos.iterable.IterableOf;
+
+/**
+ * Logical exclusive or.
+
+ * For no elements in {@code origin} it returns {@link True}, otherwise
+ * returns {@link True} only when an odd number of elements have value true.
+ *
+ * <p>This class could be also used for matching multiple boolean
+ * expressions:</p>
+ *
+ * {@code
+ * new Xor(
+ *    new True(),
+ *    new True(),
+ *    new True()
+ * ).value(); // the result is true
+ *
+ * new Xor(
+ *    new True(),
+ *    new False(),
+ *    new True()
+ * ).value(); // the result is false
+ * }
+ *
+ * <p>This class implements {@link Scalar}, which throws a checked
+ * {@link Exception}. This may not be convenient in many cases. To make
+ * it more convenient and get rid of the checked exception you can
+ * use the {@link Unchecked} decorator. Or you may use
+ * {@link IoChecked} to wrap it in an IOException.</p>
+ *
+ * <p>There is no thread-safety guarantee.
+ *
+ * @see Unchecked
+ * @see IoChecked
+ * @since 0.49
+ */
+public final class Xor implements Scalar<Boolean> {
+
+    /**
+     * The iterable.
+     */
+    private final Iterable<Scalar<Boolean>> origin;
+
+    /**
+     * Ctor.
+     * @param scalar The Scalar.
+     */
+    @SafeVarargs
+    public Xor(final Scalar<Boolean>... scalar) {
+        this(new IterableOf<>(scalar));
+    }
+
+    /**
+     * Ctor.
+     * @param iterable The iterable.
+     */
+    public Xor(final Iterable<Scalar<Boolean>> iterable) {
+        this.origin = iterable;
+    }
+
+    @Override
+    public Boolean value() throws Exception {
+        return new Ternary<>(
+            new LengthOf(this.origin).value() > 0,
+            new Reduced<Boolean>((a, b) -> a ^ b, this.origin),
+            new True()
```

Comment:
@kokodyn why is xor true if the list is empty? I would have said false, but purely intuitively, because there is nothing, so it can't be true… maybe you have some sources that recommends to do that?

Reply:
Xor for empty list should be false. This is correct answer.